examples/CustomCollectionView-Swift/Sample/MosaicCollectionViewLayout.swift

```diff
@@ -0,0 +1,254 @@
+}
+
+class MosaicCollectionViewLayout: UICollectionViewFlowLayout {
+  var numberOfColumns: Int
+  var columnSpacing: CGFloat
+  var _sectionInset: UIEdgeInsets
+  var interItemSpacing: UIEdgeInsets
+  var headerHeight: CGFloat
+  var _columnHeights: [[CGFloat]]?
+  var _itemAttributes: [[UICollectionViewLayoutAttributes]]?
+  var _headerAttributes: [UICollectionViewLayoutAttributes]?
+  var _allAttributes: [UICollectionViewLayoutAttributes]?
+  
+  required override init() {
+    self.numberOfColumns = 2
+    self.columnSpacing = 10.0
+    self.headerHeight = 44.0 //viewcontroller
+    self._sectionInset = UIEdgeInsetsMake(10.0, 10.0, 10.0, 10.0)
+    self.interItemSpacing = UIEdgeInsetsMake(10.0, 0, 10.0, 0)
+    super.init()
+    self.scrollDirection = .vertical
+  }
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  public var delegate : MosaicCollectionViewLayoutDelegate?
+  
+  override func prepare() {
+    super.prepare()
+    
+    _itemAttributes = []
+    _allAttributes = []
+    _headerAttributes = []
+    _columnHeights = []
+    
+    var top: CGFloat = 0
+    
+    let numberOfSections: NSInteger = self.collectionView!.numberOfSections
+    
+    for section in 0 ..< numberOfSections {
+      let numberOfItems = self.collectionView?.numberOfItems(inSection: section)
+      
+      top += _sectionInset.top
+      
+      if (headerHeight > 0) {
+        let headerSize: CGSize = self._headerSizeForSection(section: section)
+        
+        let attributes: UICollectionViewLayoutAttributes = UICollectionViewLayoutAttributes.init(forSupplementaryViewOfKind: UICollectionElementKindSectionHeader, with: NSIndexPath.init(row: 0, section: section) as IndexPath)
+        
+        attributes.frame = CGRect.init(x: _sectionInset.left, y: top, width: headerSize.width, height: headerSize.height)
+        _headerAttributes?.append(attributes)
+        _allAttributes?.append(attributes)
+        top = attributes.frame.maxY
+      }
+      
+      _columnHeights?.append([]) //Adding new Section
+      for _ in 0 ..< self.numberOfColumns {
+        self._columnHeights?[section].append(top)
+      }
+      
+      let columnWidth: CGFloat = self._columnWidthForSection(section: section)
+      _itemAttributes?.append([])
+      for idx in 0 ..< numberOfItems! {
+        let columnIndex: Int = self._shortestColumnIndexInSection(section: section)
+        let indexPath = IndexPath.init(item: idx, section: section)
+        
+        let itemSize: CGSize = self._itemSizeAtIndexPath(indexPath: indexPath);
+        let xOffset: CGFloat = _sectionInset.left + (columnWidth + columnSpacing) * CGFloat(columnIndex)
+        let yOffset: CGFloat = _columnHeights![section][columnIndex]
+        
+        let attributes: UICollectionViewLayoutAttributes = UICollectionViewLayoutAttributes.init(forCellWith: indexPath)
```

Comment:
You don't need `let attributes: UICollectionViewLayoutAttributes =`, just use `let attributes =`, the type will be automatically inferred by the equality.
You don't need the .init(...) you can just use (...)